include/seqan3/alphabet/cigar/cigar.hpp

```diff
@@ -0,0 +1,409 @@
+// ------------------------------------------------------------------
+
+namespace std
+{
+
+/*!\brief std::tuple_size overload.
+ * [metafunction specialisation for seqan3::cigar]
+ */
+template<typename length_type>
+struct tuple_size<seqan3::cigar<length_type>>
+{
+    static constexpr size_t value = 2; //!< Composition of seqan3::cigar_op and length value.
+};
+
+/*!\brief std::tuple_element overload.
+ * [metafunction specialisation for seqan3::cigar]
+ */
+template<size_t elem_no,
+         typename length_type>
+struct tuple_element<elem_no, seqan3::cigar<length_type>>
+    : tuple_element<elem_no, std::tuple<seqan3::cigar_op, length_type>>
+{};
+
+} // namespace std
+
+
+// ------------------------------------------------------------------
+// create cigar_vector out of alignment
+// ------------------------------------------------------------------
+
+namespace seqan3
+{
+
+/*!\brief Compares two aligned sequence values and returns their seqan3::cigar_op.
+ * \ingroup aligned_sequence
+ * \relatesalso cigar
+ * \tparam reference_char_type Must be equality comparable to seqan3::gap.
+ * \tparam query_char_type     Must be equality comparable to seqan3::gap.
+ * \param  reference_char      The aligned character of the reference to compare.
+ * \param  query_char          The aligned character of the query to compare.
+ * \param  extended_cigar      Whether to print the extended cigar alphabet or not. See seqan3::cigar_op.
+ * \returns A seqan3::cigar_op representing the alignment operation between the
+ *          two values.
+ *
+ * \details
+ *
+ * \attention Note that seqan3::cigar elements (respectively by their
+ *            seqan3::cigar_op) are always related to on one of the two sequences
+ *            in a pairwise alignment. In this case, the resulting
+ *            operation is related to the \p query_char.
+ *
+ * ### Example:
+ *
+ * The following alignment column shows the reference char ('C') on top and a
+ * gap for the query char at the bottom.
+ * ```
+ * ... C ...
+ *     |
+ * ... - ...
+ * ```
+ * In this case, the function seqan3::compare_aligned_values will return
+ * seqan3::cigar_op::D, since the query char is "deleted".
+ *
+ * The next alignment column shows the reference char ('C') on top and a
+ * query char ('G') at the bottom.
+ * ```
+ * ... C ...
+ *     |
+ * ... G ...
+ * ```
+ * In this case, the function seqan3::compare_aligned_values will return
+ * seqan3::cigar_op::M, for the basic cigar the two bases are aligned, while
+ * in the extended CIGAR alphabet (\p extended_cigar = `true`) the function
+ * will return an seqan3::cigar_op::X since the bases are aligned but are not
+ * equal.
+ */
+template<typename reference_char_type, typename query_char_type>
+//!\cond
+    requires std::EqualityComparableWith<std::remove_reference_t<reference_char_type>, gap> &&
+             std::EqualityComparableWith<std::remove_reference_t<query_char_type>, gap>
+//!\endcond
```

Comment:
you can also use shorter syntax here and don't forget the space between `template <typename`

Reply:
shouldn't this function maybe go into detail?